src/charm.py

```diff
@@ -28,7 +30,7 @@ class MySQLCharm(CharmBase):
 
     def __init__(self, *args):
         super().__init__(*args)
-        self._stored.set_default(mysql_setup={})
+        self._stored.set_default(mysql_setup={"MYSQL_ROOT_PASSWORD": False})
         self._stored.set_default(mysql_initialized=False)
```

Comment:
Haven't tried it myself, but maybe emitting a custom 'unit ready' event instead would be better?

Reply:
I will take a look at this suggestion, thanks!